python/lsst/ts/mtdome/power_management/power_management_handler.py

```diff
@@ -0,0 +1,486 @@
+                missing_pmm.add(power_management_mode)
+        if len(missing_pmm) > 0:
+            raise RuntimeError(
+                "The following PowerManagementModes are missing from the configuration: "
+                + ", ".join([pmm.name for pmm in missing_pmm])
+            )
+
+        self.command_queue: asyncio.PriorityQueue = asyncio.PriorityQueue()
+        self.command_priorities = command_priorities
+        self.power_management_mode = PowerManagementMode.NO_POWER_MANAGEMENT
+        self.log = log.getChild(type(self).__name__)
+
+    async def schedule_command(self, command: ScheduledCommand) -> None:
+        """Schedule a command to be issued.
+
+        The priority of the command is looked up in the command_priorities for
+        the current PowerManagementMode. If the command is not present then the
+        DEFAULT_PRIORITY is used.
+
+        Parameters
+        ----------
+        command : `ScheduledCommand`
+            The command and its parameters to schedule.
+        """
+        priority = HIGH_PRIOTITY
+        command_priorities_to_use = self.command_priorities[self.power_management_mode]
+        if command.command in command_priorities_to_use:
+            priority = command_priorities_to_use[command.command]
+        await self.command_queue.put((priority, command))
+
+    async def get_next_command(
+        self, current_power_draw: dict[str, float]
+    ) -> ScheduledCommand | None:
+        """Get the next command to be issued, or None if no commands currently
+        can be issued or are scheduled.
+
+        This method delegates getting the next command to the specific method
+        for the current PowerManagementMode.
+
+        Parameters
+        ----------
+        current_power_draw : `dict`[`LlcName`, `float`]
+            Dict of the current power draw [W] for the subsystems, or Lower
+            Level Components.
+
+        Returns
+        -------
+        ScheduledCommand | None
+            The next command to be issued, or None if no commands currently can
+            be executed or are scheduled.
+        """
+        if self.command_queue.empty():
+            return None
+
+        match self.power_management_mode:
+            case PowerManagementMode.OPERATIONS:
+                scheduled_command = await self.get_next_command_operations(
+                    current_power_draw
+                )
+            case PowerManagementMode.MAINTENANCE:
+                scheduled_command = await self.get_next_command_maintenance(
+                    current_power_draw
+                )
+            case PowerManagementMode.EMERGENCY:
+                scheduled_command = await self.get_next_command_emergency(
+                    current_power_draw
+                )
+            case PowerManagementMode.NO_POWER_MANAGEMENT | _:
+                _, scheduled_command = await self.command_queue.get()
+        return scheduled_command
+
+    async def get_next_command_operations(
```

Comment:
If I understood this correctly, this method is not only "getting" the next command but also "issuing" the command. If this is correct, I think it might be a bit misleading to call this method "get_<something>". Maybe call it `execute_next_command_operations`?

Reply:
Good point but it actually doesn't issue the command. That is done by the cRIO. So `get_` is correct for this and the other two PowerManagementMode-specific methods. I have renamed the `issue_scheduled_command_or_stop_commands` to `generic_get_scheduled_command_or_stop_commands` to reflect that there too.